diffkemp/simpll/passes/VarValueDependencySlicer.cpp

```diff
@@ -0,0 +1,123 @@
+//== VarValueDependencySlicer.cpp-Removing code based on value of variable ==//
+//
+//        SimpLL - Program simplifier for analysis of semantic difference    //
+//
+// This file is published under Apache 2.0 license. See LICENSE for details.
+// Author:
+//===---------------------------------------------------------------------===//
+///
+/// \file
+/// This file contains the implementation of the VarValueDependencySlicer pass.
+/// The pass slice the program to a value of a global variable.
+///
+//===--------------------------------------------------------------------===//
+
+#include "VarValueDependencySlicer.h"
+#include "DebugInfo.h"
+#include <Config.h>
+#include <Utils.h>
+#include <llvm/IR/Constant.h>
+#include <llvm/IR/Constants.h>
+#include <llvm/IR/GlobalValue.h>
+#include <llvm/IR/GlobalVariable.h>
+#include <llvm/IR/InstrTypes.h>
+#include <llvm/IR/Instructions.h>
+#include <llvm/Support/FileSystem.h>
+#include <llvm/Support/raw_ostream.h>
+#include <utility>
+
+using namespace llvm;
+
+PreservedAnalyses VarValueDependencySlicer::run(Module &Mod,
+                                                ModuleAnalysisManager &mam,
+                                                GlobalVariable *Var,
+                                                std::string VarValue) {
+    Constant *initializer, *newConst;
+    SmallVector<Instruction *, 128> *listOfLoadInstructions =
+            new SmallVector<Instruction *, 128>();
+    SmallVector<std::pair<Instruction *, int>, 128>
+            *listOfLoadInstructionUsers =
+                    new SmallVector<std::pair<Instruction *, int>, 128>();
+
+    if (Var->hasInitializer())
+        initializer = Var->getInitializer();
+    else
+        return PreservedAnalyses::all();
+
+    if (VarValue != "default") {
+        newConst = getConstantFromString(initializer, VarValue);
+
+        if (newConst == NULL)
+            return PreservedAnalyses::none();
+    } else
+        newConst = initializer;
```

Comment:
I think that it would make more sense to move this (getting appropriate `Constant` from a `string`) to parsing of CLI arguments and that this pass should accept `Constant *` instead of `std::string` for parameter `VarValue`.